src/ngraph/pass/concat_fusion.cpp

```diff
@@ -99,22 +99,31 @@ void pass::ConcatElimination::construct_concat_elimination()
             << "concat_elimination: In callback for construct_concat_elimination against node = "
             << m.get_match_root()->get_name();
         auto pattern_map = m.get_pattern_map();
-        auto op = pattern_map[op_label];
+        auto concat_input_op = pattern_map[op_label];
```

Comment:
You want `pattern_value_map` here; `pattern_map` is for back-compatibility with GOE-based patterns, so 
`auto concat_input = m.get_pattern_value_map()[op_label];`
`pattern_value_map` will give you the correct output corresponding to the label. And then you'll want
`concat.output(0).replace(concat_input);`